### PR TITLE
refactor: use BitcoinModule to fetch BTC balances

### PR DESCRIPTION
**Pull request by meeh0w**
## Description
* [CP-9001](https://ava-labs.atlassian.net/browse/CP-9001)
* Ongoing effort to integrate the `BitcoinModule`, some related PRs are:
    * #7 
    * #8

## Changes
* Replaces direct `BitcoinProvider` + `TokenPricesService` calls with a single call to the `BitcoinModule`
* Drops `BN` from Bitcoin balances
    * I added some helper methods for the transition period inside `background/services/balances/models.ts` file, we'll be able to remove them once all VM modules are integrated

## Testing
* Run sanity checks on the app
* Perform some transactions (send / swap / bridge)
* There should be no functional changes (slight changes to the balance formatting are okay, however I did not notice any)
    * This is because `balanceDisplayValue` / `balanceCurrencyDisplayValue` are returned by the modules and they use [`TokenUnit`](https://github.com/ava-labs/avalanche-sdks/blob/dd4e59e8d3c70b5922424a81d5caf1d95ba26172/packages/utils/src/tokenUnit.ts) for calculations / formatting

## Screenshots:
_N/A_

## Checklist for the author
- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

File: src/background/services/balances/utils/isTokenWithBalanceAVM.test.ts [hunk 1]
```diff
@@ -42,10 +45,12 @@ describe('src/background/services/balances/utils/isTokenWithBalanceAVM.ts', () =
   };
 
   const tokenWithBalanceBTC: TokenWithBalanceBTC = {
-    type: TokenType.NATIVE,
+    type: VMModulesTokenType.NATIVE,
     name: 'Avalanche',
     symbol: 'AVAX',
-    balance: new BN(1),
+    coingeckoId: 'btc',
```
Comment (bferenc) [inline src/background/services/balances/utils/isTokenWithBalanceAVM.test.ts:51]: 🤓 
```suggestion
    coingeckoId: 'avax',
```

File: src/background/services/balances/utils/isTokenWithBalancePVM.test.ts [hunk 2]
```diff
@@ -45,7 +45,9 @@ describe('src/background/services/balances/utils/isTokenWithBalancePVM.ts', () =
     type: TokenType.NATIVE,
     name: 'Avalanche',
     symbol: 'AVAX',
-    balance: new BN(1),
+    balance: 1n,
+    balanceDisplayValue: '0.00000001',
+    coingeckoId: 'btc',
```
Comment (bferenc) [inline src/background/services/balances/utils/isTokenWithBalancePVM.test.ts:50]: 🤓 
```suggestion
    coingeckoId: 'avax',
```

**Review by bferenc — APPROVED** [review]
Nice! Also tested with different tx types as advised and everything was working properly 🙌

File: src/background/services/balances/utils/isTokenWithBalanceAVM.test.ts (hunk 1 above)
Comment (meeh0w) [reply src/background/services/balances/utils/isTokenWithBalanceAVM.test.ts:51]: 🍝

Comment (meeh0w) [reply src/background/services/balances/utils/isTokenWithBalanceAVM.test.ts]: 1dce1c034

File: src/background/services/balances/utils/isTokenWithBalancePVM.test.ts (hunk 2 above)
Comment (meeh0w) [reply src/background/services/balances/utils/isTokenWithBalancePVM.test.ts]: 1dce1c034

**Review by bferenc — DISMISSED** [review]

**Review by gergelylovas — DISMISSED** [review]

**Review by bferenc — APPROVED** [review]

**Review by gergelylovas — APPROVED** [review]

**Pull request merged**